dlt/load/utils.py

```diff
@@ -128,12 +140,26 @@ def _init_dataset_and_update_schema(
     update_tables: Iterable[str],
     truncate_tables: Iterable[str] = None,
     staging_info: bool = False,
+    drop_tables: Optional[List[TTableSchema]] = None,
 ) -> TSchemaTables:
     staging_text = "for staging dataset" if staging_info else ""
     logger.info(
         f"Client for {job_client.config.destination_type} will start initialize storage"
         f" {staging_text}"
     )
+    if drop_tables:
+        old_schema = job_client.schema
```

Comment:
we already dropped the tables from the `old_schema` right? so if we drop them again this means they were added from the data. (or we have a bug that we do not drop this tables effectively).

maybe we do not need to drop anything here? we should have here a schema in a final state

Reply:
I think at this point the dropped tables are back in `old_schema` if the same resources are being loaded again.  
So I had to do this and replace stored schema in `_dlt_version` so that during schema update it correctly detects these tables don't exist and they get re-created.   
It's a little convoluted though for sure, maybe there's a better way.

Reply:
radical solution would be to:
1. add a flag to `update_stored_schema` that always forces to re-check the schema and create/update existing tables
2. remove `replace_schema` from `drop_tables`
3. remove `_drop_destination_tables` from drop command and always create empty package (just modified state, schema and tables that needs to be dropped and truncated and do this here)

Reply:
> add a flag to update_stored_schema that always forces to re-check the schema and create/update existing tables
This would have to run every time, right? We don't know if a table was dropped in some previous load.

Reply:
Potentially solved it with this: https://github.com/dlt-hub/dlt/pull/1063/commits/8990f114e9664b16fad55d6d2eccc73c217af5d2  
`replace_schema` is not needed now, but I check both `version_hash` and `version` when getting stored destination schema, so we are not comparing with the "pre-drop" schema. Maybe version number or previous hashes or something could be part of the hash so it's simpler?   
Are there any downsides to this?